File: frontend/src/types/answer.ts
export type SuggestedAnswer = {
  id: number;
  question_id: number;
  answer_text: string;
  star_situation: string | null;
  star_task: string | null;
  star_action: string | null;
  star_result: string | null;
  created_at: string;
};

File: backend/app/schemas/answer.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SuggestedAnswerResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    question_id: int
    answer_text: str
    star_situation: str | None = None
    star_task: str | None = None
    star_action: str | None = None
    star_result: str | None = None
    created_at: datetime
